uiowa/uiowa
# [ITS] Outage and Degradation component

Resolves #7197 
Resolves #7216 





# How to test

```
ddev blt ds --site=its.uiowa.edu && ddev drush @its.local uli && ddev drush @its.local uli
```
^ Note the added cache clear, which is necessary to get it to recognize the `ItsAlert` over the standard `Alert`.

Visit https://its.uiowa.ddev.site/admin/structure/taxonomy/manage/alert_categories/overview and try to delete any of the four category terms, and see that the delete button is disabled and a suitable message is displayed.
Add a new category taxonomy term and check that you are able to delete it. (Go ahead and delete it to keep things cleaner when creating alerts later.)

Create/edit a page and add the "Outages and Degradations", "Planned Maintenance", and "Service announcements" list blocks (feel free to test in whatever layout you'd like, but https://sandbox.prod.drupal.uiowa.edu/mockup-its-alerts-landing-7 can be used as a reference mockup to recreate).

Check the no results messages appear (may have to delete the existing "Test Alert" outage).

Add some alerts with the various categories, and various states of "open" or "closed" (eg an "open" alert is one that is ongoing, or has a defined end date later than right now). Check out the views and see that they include any "open" alerts, but not any "closed" ones.

Add a new alert (outage/degradation, maintenance, or service announcement category) and refresh the page--see that the views refresh.
Delete an existing alert and refresh the page--see that the views refresh.

Create an alert with an end time close to the current time. Load the page with the views blocks and see that it appears in the appropriate view.

Test any other instances you can think of where an alert should appear or drop off of the views, and check that the caching handles it appropriately (if an alert is not created/edited, cron will need to be used)

```
ddev drush @its.local cron`
```

After cron runs, reload the page and see that the now closed alert no longer appears in the view.



## Iowa Now bug fix

Delete confirmation got moved to a modal, and so the message that previously got added as a warning isn't visible until after canceling (and depending on your actions, sometimes not viewed until moving to a different page, at which point all warnings appear). This moves it from a warning and instead replaces the modal's description with our own.

```
ddev blt ds --site=now.uiowa.edu && ddev drush @now.local uli /admin/structure/taxonomy/manage/tags/overview
```
Try to delete 'achievements' and see that you can't, and are given an accurate message that you cannot
Try to delete another term, and see that you can
